src/FunctionApps/python/GenerateCSVs/elr.py

```diff
@@ -17,12 +17,18 @@ def extract_loinc_lab(observation: dict) -> List[str]:
         code = observation["code"]["coding"][0]
     except KeyError:
         return []
-    if "loinc" in code["system"] and "valueCodeableConcept" in observation:
+    if "loinc" in code["system"]:
         obs_date = observation.get("effectiveDateTime")
-        try:
-            result = observation["valueCodeableConcept"]["coding"][0]["display"]
-        except KeyError:
-            result = ""
+        if "valueCodeableConcept" in observation:
+            try:
+                result = observation["valueCodeableConcept"]["coding"][0]["display"]
+            except KeyError:
+                result = ""
+        elif "valueString" in observation:
+            try:
+                result = observation["valueString"]
+            except KeyError:
+                result = ""
```

Comment:
Do we need a:
```
else:
     result = ""
```
at the end of the if block so there is always a value of result to return?

Reply:
No, because there's a check on the length of the array created here (line 84 in `ecr.py` and line 51 in `elr.py`) that uses length to determine whether we should write the result to a row. Essentially the defensive coding bit is just moved to the caller instead of this function.

Reply:
Wait nevermind I thought you were talking about something else. Yes we absolutely do, added now.